Realm/Realm.Weaver/Analytics/Analytics.cs

```diff
@@ -0,0 +1,846 @@
+                {
+                    if (!(member is PropertyDefinition property))
+                    {
+                        return false;
+                    }
+
+                    if (((GenericInstanceType)property.PropertyType).GenericArguments[1].IsPrimitive)
+                    {
+                        featureDict[PrimitiveDictionary] = 1;
+                    }
+                    else
+                    {
+                        featureDict[ReferenceDictionary] = 1;
+                    }
+
+                    return featureDict[PrimitiveDictionary] == 1 &&
+                        featureDict[ReferenceDictionary] == 1;
+                }
+            },
+            {
+                "ISet`1", (member, featureDict, references) =>
+                {
+                    if (!(member is PropertyDefinition property))
+                    {
+                        return false;
+                    }
+
+                    if (((GenericInstanceType)property.PropertyType).GenericArguments[0].IsPrimitive)
+                    {
+                        featureDict[PrimitiveSet] = 1;
+                    }
+                    else
+                    {
+                        featureDict[ReferenceSet] = 1;
+                    }
+
+                    return featureDict[PrimitiveSet] == 1 &&
+                        featureDict[ReferenceSet] == 1;
+                }
+            },
+            {
+                "RealmInteger`1", (member, featureDict, references) =>
+                {
+                    featureDict[RealmInteger] = 1;
+                    return true;
+                }
+            },
+        };
+
+        #endregion
+
+        private readonly Config _config;
+
+        internal Analytics(Config config, ImportedReferences references, ILogger logger)
+        {
+            _references = references;
+            _config = config;
+            _logger = logger;
+        }
+
+        private void AnalyzeRealmClass(TypeDefinition type)
+        {
+            Func<IMemberDefinition, Dictionary<string, byte>, ImportedReferences, bool> featureFunc = null;
+
+            if (_classAnalysisSetters.TryGetValue(nameof(IAsymmetricObject), out featureFunc) &&
+                (type.IsIAsymmetricObjectImplementor(_references) || type.IsAsymmetricObjectDescendant(_references)) &&
+                featureFunc.Invoke(type, _realmFeaturesToAnalyse, _references))
+            {
+                _classAnalysisSetters.Remove(nameof(IAsymmetricObject));
+            }
+            else if (_classAnalysisSetters.TryGetValue(nameof(IEmbeddedObject), out featureFunc) &&
+                (type.IsIEmbeddedObjectImplementor(_references) || type.IsEmbeddedObjectDescendant(_references)) &&
+                featureFunc.Invoke(type, _realmFeaturesToAnalyse, _references))
+            {
+                _classAnalysisSetters.Remove(nameof(IEmbeddedObject));
+            }
+
+            foreach (var property in type.Properties)
```

Comment:
Are we getting all properties here, including the ones that will not be part of the schema?

Reply:
Yes. It looks at the assembly whle the weaving task is carried out on that same assembly.
I guess you're correct in suggesting of not analyzing the objects that don't make it into the schema. I'll look at how to piggyback what we already have in `WeaveSchema`.

Reply:
I'm mostly worried about properties that are not in the schema because developers could have other properties that are not persisted. The objects that are not in the schema...maybe it's slightly less important

Reply:
I worked this out and now only properties that are part of the schema are checked. You can resolve this if you are satisfied, let me know otherwise.